### PR TITLE
[SPARK-14597] [Streaming] Streaming Listener timing metrics should include time spent in JobGenerator's graph.generateJobs

### PR DESCRIPTION
**Pull request by agsachin**
## What changes were proposed in this pull request?

I have added the complete information on the jira https://issues.apache.org/jira/browse/SPARK-14597
## How was this patch tested?

not tested yet to discuss and finalize the approach.

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala [hunk 1]
```diff
@@ -241,6 +242,7 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
 
   /** Generate jobs and perform checkpoint for the given `time`.  */
   private def generateJobs(time: Time) {
+    listenerBus.post(StreamingListenerBatchGenerateStarted(clock.getTimeMillis()))
```
Comment (mariobriggs) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala]: The StreamingListenerBatchGenerateStarted and StreamingListenerBatchGenerateCompleted should report the batchTime information as well. If not the end-user will not be able to correlate this processingTime to the relevant batch.

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala [hunk 2]
```diff
@@ -30,6 +30,12 @@ import org.apache.spark.util.Distribution
 sealed trait StreamingListenerEvent
 
 @DeveloperApi
+case class StreamingListenerBatchGenerateStarted(time: Long) extends StreamingListenerEvent
+
+@DeveloperApi
+case class StreamingListenerBatchGenerateCompleted(time: Long) extends StreamingListenerEvent
```
Comment (mariobriggs) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListener.scala]: you probably want the StreamingListenerBatchGenerateStarted & StreamingListenerBatchGenerateCompleted classes to have a BatchGenerateInfo class that has following fields & methods
   batchTime
   processingStartTime
   processingEndTime
   def processingDelay() : Long

**Comment by mariobriggs** [comment]
It also makes sense to add test cases for this one

**Comment by agsachin** [comment]
hey closed this thread as we are moving towards approach 2 explained in the jira https://issues.apache.org/jira/browse/SPARK-14597

**Pull request closed without merging**